File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.py
# Last updated: 6/25/2026, 9:17:49 AM
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        sp= haystack.split(needle)
        return len(sp[0]) if len(sp)>1 else -1
        

File: 28-FindTheIndexOfTheFirstOccurrenceInAString/28-FindTheIndexOfTheFirstOccurrenceInAString.cpp
// Last updated: 6/25/2026, 9:17:54 AM
class Solution {
public:
    int strStr(string haystack, string needle) {
        if(haystack.size()<needle.size()) return -1;
        for(int i=0; i<haystack.size()-needle.size()+1; i++){
            if(haystack[i] == needle[0]){
                for(int j=0; j<needle.size(); j++){
                    if(needle[j] != haystack[i+j]) break;
                    if(j==needle.size()-1) return i;
                }
            }
        }
        return -1;
    }
};